llvm/lib/Target/AMDGPU/SILowerSGPRSpills.cpp

```diff
@@ -278,7 +278,10 @@ void SILowerSGPRSpills::extendWWMVirtRegLiveness(MachineFunction &MF,
   for (auto Reg : MFI->getSGPRSpillVGPRs()) {
     for (MachineBasicBlock *SaveBlock : SaveBlocks) {
       MachineBasicBlock::iterator InsertBefore = SaveBlock->begin();
-      auto MIB = BuildMI(*SaveBlock, *InsertBefore, InsertBefore->getDebugLoc(),
+      DebugLoc DL = InsertBefore != SaveBlock->end()
+                        ? InsertBefore->getDebugLoc()
+                        : DebugLoc();
```

Comment:
This looks like a job for `MachineBasicBlock::findDebugLoc` (not that I've ever used it myself):
```suggestion
      DebugLoc DL = SaveBlock->findDebugLoc(InsertBefore);
```

Reply:
Thanks - findDebugLoc is much better.